packages/react-native-reanimated/src/logger/logger.ts

```diff
@@ -55,27 +78,84 @@ export function logToLogBoxAndConsole(data: LogData) {
 }
 
 /**
- * Replaces the default log function with a custom implementation.
+ * Registers the logger configuration.
+ * use it only for Worklet runtimes.
+ *
+ * @param config - The config to register.
+ */
+export function registerLoggerConfig(config: LoggerConfigInternal) {
+  'worklet';
+  if (!_WORKLET) {
+    throw new Error(
+      '[Reanimated] registerLoggerConfig() must be called on Worklet runtime'
+    );
+  }
+  (global as Record<string, unknown>).__reanimatedLoggerConfig = config;
+}
+
+/**
+ * Updates logger configuration.
  *
- * @param logFunction - The custom log function.
+ * @param options - The new logger configuration to apply.
+ *   - level: The minimum log level to display.
+ *   - strict: Whether to log warnings and errors that are not strict.
+ *    Defaults to false.
+ *   - logFunction: The function to use for logging.
  */
-export function replaceLoggerImplementation(
-  logFunction: (data: LogData) => void
+export function updateLoggerConfig(options?: Partial<LoggerConfigInternal>) {
+  'worklet';
+  // Clone the config in the worklet runtime thus the object sent to
+  // the worklet runtime is immutable
+  const config = _WORKLET
+    ? {
+        ...(((global as Record<string, unknown>)
+          .__reanimatedLoggerConfig as LoggerConfigInternal) ?? {}),
+      }
+    : __reanimatedLoggerConfig;
```

Comment:
I don't get it, could you elaborate?

Reply:
Well, I cannot just assign `__reanimatedLoggerConfig` to the `context` variable as `__reanimatedLoggerConfig` is not assigned to `global` variable on the Worklet runtime yet, thus, trying to access `__reanimatedLoggerConfig` in the same way as I do in the `handleLog` (`const config = __reanimatedLoggerConfig;`) will crashes the app.

The result was to access it via the `global` variable, which is defined. We will get `undefined` for the `global.__reanimatedLoggerConfig` if `__reanimatedLoggerConfig` hadn't been registered in the global scope, so in that case, I am returning the empty object as a fallback, which will be filled in below in this function.

Destructuring is no longer necessary and I forgot to remove that, so this implementation can be simplified to:

```tsx
  const config = _WORKLET
    ? ((global as Record<string, unknown>)
        .__reanimatedLoggerConfig as LoggerConfigInternal) ?? {}
    : __reanimatedLoggerConfig;
```